models/user.go

```diff
@@ -395,10 +395,20 @@ func hashPassword(passwd, salt, algo string) string {
 	return fmt.Sprintf("%x", tempPasswd)
 }
 
-// HashPassword hashes a password using the algorithm defined in the config value of PASSWORD_HASH_ALGO.
-func (u *User) HashPassword(passwd string) {
+// SetPassword hashes a password using the algorithm defined in the config value of PASSWORD_HASH_ALGO
+// change passwd, salt and passwd_hash_algo fields
+func (u *User) SetPassword(passwd string) (err error) {
+	if len(passwd) == 0 {
+		return fmt.Errorf("no passord to be set")
```

Comment:
This is going to break creation of external users

Reply:
why is SetPassword osed on external Users?!?

Reply:
It's in:

`func CreateUser(u *User) (err error) {`

Reply:
right this is independent ...

Reply:
@zeripath fixed - still I think before mrege/lgtm we neet to test this a bit ...

Reply:
(I'm worried about edge-cases)

Reply:
Could do something like test the LoginSource and if it's != 0 then assert the password has to be something non-empty - but setting password to empty could just be used to disable login locally I guess.

Reply:
Oauth2 account creation works :)
local account too :rocket: 

not tested: LDAP, SMTP, ...

Reply:
I think we should allow setPassword to set to "" - logic disallowing users to set their own password to "" belongs at the router level.

The reasoning here is that setPassword appears to be the only way to change the password - and therefore disable local login.

Reply:
hmm so if it is  "" we just clear all pwd fileds and it should be fine ...